fe/fe-core/src/main/java/com/starrocks/connector/parser/trino/Trino2SRFunctionCallTransformer.java

```diff
@@ -326,7 +349,12 @@ private static void registerFunctionTransformerWithVarArgs(String trinoFnName, S
 
     private static void registerFunctionTransformer(String trinoFnName, int trinoFnArgNums,
                                                     FunctionCallExpr starRocksFunctionCall) {
-        FunctionCallTransformer transformer = new FunctionCallTransformer(starRocksFunctionCall, trinoFnArgNums);
+        FunctionCallTransformer transformer;
+        if (trinoFnArgNums == 0) {
+            transformer = new FunctionCallTransformer(starRocksFunctionCall, false);
```

Comment:
when trinoFnArgNums == 0, what's the difference 
```suggestion
            transformer = new FunctionCallTransformer(starRocksFunctionCall,trinoFnArgNums);
```

Reply:
I made this change here because the constructor of FunctionCallTransformer handles false and 0 differently.

```
    public FunctionCallTransformer(FunctionCallExpr targetCall, int argNums) {
        this(targetCall, false, argNums);
    }

    public FunctionCallTransformer(FunctionCallExpr targetCall, boolean variableArgument) {
        this(targetCall, variableArgument, 0);
    }

    private FunctionCallTransformer(FunctionCallExpr targetCall, boolean variableArgument, int argNums) {
        this.targetCall = targetCall;
        this.argNums = argNums;
        this.variableArgument = variableArgument;
        if (variableArgument) {
            this.placeholderExprs = Lists.newArrayList();
        } else {
            this.placeholderExprs = Arrays.asList(new PlaceholderExpr[argNums]);
        }
        init();
    }
```

If you do not explicitly set variableArgument to false, the array will be out of bounds.

Reply:
This ctor already set variableArgument to false? why not just use this 
``` 
public FunctionCallTransformer(FunctionCallExpr targetCall, int argNums) {
        this(targetCall, false, argNums);
 }
```

Reply:
Yeah, you are right. 😓 
I should be confused ..... here
`` `if (variableArgument) {` `` `
I test this with my computer... wait a moment

Reply:
Already modified and pushed according to your opinions, the code test is normal, thank you！

@Youngwb

Reply:
@happut  plz use the original, there is no need to modify
![image](https://github.com/StarRocks/starrocks/assets/9495145/617f84c5-4555-4dc7-8d74-9dc420c771be)

Reply:
Has been modified and pushed again, you can see the latest Commit
@Youngwb

Reply:
Execuse me, again.  Mentor, I saw the 'FE Sonarcloud Check (pull_request) Failing after 2m', Should I merge the latest code of branch main of starrocks/starrocks to solved this check error?

Reply:
@happut 
![image](https://github.com/StarRocks/starrocks/assets/9495145/86eaface-4cda-427d-8cd8-ca2ffce7c832)
you can rebase main branch and check this

Reply:
Is there any problem with the Main branch? After I rebase, I was reported an error, I couldn't compile.
![image](https://github.com/StarRocks/starrocks/assets/2594794/9b169d69-732a-4017-9539-b385da848918)

Reply:
This is the error.
![image](https://github.com/StarRocks/starrocks/assets/2594794/d0b1d443-a7f6-4de5-ad4a-b96c2cb561a1)

I think it should be this ?
```
        this.generatedColumnExpr = column.generatedColumnExpr;
```

Reply:
sorry , there is compilation problem in main branch, after we fix it, you can rebase it again

Reply:
https://github.com/StarRocks/starrocks/pull/47522 this PR fixed